Feature/ light groups

boberfly opened the pull request:
Adding light group support
- Patch derives from https://developer.blender.org/D3607
- Addresses https://github.com/tangent-opensource/coreBlackbird/issues/33
- Blender exporter needs updating still

boberfly commented:
@skwerner I didn't update the blender exporter code, should we do this?

skwerner commented:
How urgent is this one? Can this wait a couple of days until we have a 2.93 merge?

boberfly commented:
@skwerner I don't mind pushing this a bit if it's just a few days if it makes things easier to merge. Cheers!

dedoardo commented on src/kernel/kernel_types.h:
```diff
@@ -1366,6 +1372,10 @@ typedef struct KernelBackground {
 
   int use_mis;
+
+  int lightgroup;
```
Why is this not `uint`?

dedoardo replied:
Nice, it was a small detail ^^
I would have kept [this](https://github.com/tangent-opensource/coreBlackbird/pull/56/commits/50008d2273c6b155bbea474f3b541b1b9e1aed55#diff-6a7e80625845a75e7271b42cb5ec928f33d46546232377b080e32582929d9f4bR1312) `int` though, to be consistent with other passes. The rest looks ok :)

boberfly replied:
Yeah if we maintain a max lightgroups to like 32 then int is fine. I think this will need a re-visit once we get light path expressions and we need to use bitmasks again.

boberfly commented:
Hey all, the latest changes I did:
- Rebased on latest master
- Fixed any int type occurrence to uint (thanks for pointing out @dedoardo )
- Any 'lightgroups' plural is now just 'lightgroup', and removed any mention of bitfields as they are not being used now
